app/controllers/AccountManagement.scala

```diff
@@ -202,39 +200,42 @@ object ManageWeekly extends ContextLogging {
              paymentMethodIsPaymentCard: Boolean
            )(implicit
              request: Request[AnyContent],
-             resolution: TouchpointBackend.Resolution
+             resolution: TouchpointBackends.Resolution
            ): Future[Result] = {
     implicit val tpBackend = resolution.backend
     implicit val rest = tpBackend.simpleRestClient
     implicit val zuoraRest = new ZuoraRestService[Future]
     implicit val flash = request.flash
     implicit val subContext = weeklySubscription
 
-    def getRenewalPlans(account: ZuoraRestService.AccountSummary, currency: Currency) = {
-      val catalog = tpBackend.catalogService.unsafeCatalog
+    def getRenewalPlans(account: ZuoraRestService.AccountSummary, currency: Currency): Future[\/[NonEmptyList[String], List[WeeklyPlanInfo]]] = {
+      EitherT(tpBackend.catalogService.catalog).flatMap { catalog =>
```

Comment:
this function returns a future anyway so we might as well flatmap the future catalog rather than awaiting the catalog itself and then using it directly.  Then it down to the caller to wait.

This was needed because if the future times out, it throws an error.  This was not a problem before as with it being a lazy static val it seemed to retry.  When I changed touchpoint backend to be a class, it seemed to try once and then fail instantly, so I thought it's a good time as any to fix properly.

Reply:
💯